key-wallet/KEY_DERIVATION_ISSUES.md

```
@@ -0,0 +1,212 @@
+
+For non-hardened child paths (all external/internal address pool scanning), public key derivation is cryptographically sufficient. Deriving private keys here unnecessarily exposes private key material to the check function's closure scope and any future logging/error paths.
+
+**Fix:** Accept `&ExtendedPubKey` and use `derive_pub`. Watch-only wallets can use this without any private key. Callers that have a private key can pass `ExtendedPubKey::from_priv(secp, &priv_key)` once at the call site.
+
+---
+
+## Issue 7 — Provider key paths have no named constants in `dip9.rs`
+
+All other DIP9 paths are defined as `IndexConstPath` constants:
+
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)